File: node/key-exchange.js
const readline = require('readline');

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

const p = 97n
const g = 18n

rl.question('Enter your secret: ', (secret) => {
    const a = BigInt(secret.trim())
    const A = (g ** a) % p
    console.log(`Value to share: ${A}`)

    rl.question('Enter the public value from the other person: ', (answer) => {
        const B = BigInt(answer.trim())
        const simpleKey = (B ** a) % p
        console.log(`Simple key: ${simpleKey}`)

        rl.close()
    })
})

File: c-sharp/KeyExchange.cs
using System;
using System.Numerics;

namespace DiffieHellman
{
    class KeyExchange
    {
        public void Exchange()
        {
            Console.Write("Enter a secret: ");
            BigInteger p = 97;
            BigInteger g = 18;
            var a = BigInteger.Parse(Console.ReadLine().Trim());
            var A = BigInteger.ModPow(g, a, p);
            Console.WriteLine($"Value to share: {A}");

            Console.Write("Enter the public value from the other person: ");
            var B = BigInteger.Parse(Console.ReadLine().Trim());
            var simpleKey = BigInteger.ModPow(B, a, p);
            Console.WriteLine($"Simple key: {simpleKey}");
        }
    }
}
